pkg/zk/generators.go

```diff
@@ -175,13 +174,43 @@ func makeZkPodSpec(z *v1beta1.ZookeeperCluster, volumes []v1.Volume) v1.PodSpec
 		Affinity:   z.Spec.Pod.Affinity,
 		Volumes:    append(z.Spec.Volumes, volumes...),
 	}
-	if reflect.DeepEqual(v1.PodSecurityContext{}, z.Spec.Pod.SecurityContext) {
+	if z.Spec.Pod.SecurityContext != nil {
 		podSpec.SecurityContext = z.Spec.Pod.SecurityContext
 	}
 	podSpec.NodeSelector = z.Spec.Pod.NodeSelector
 	podSpec.Tolerations = z.Spec.Pod.Tolerations
 	podSpec.TerminationGracePeriodSeconds = &z.Spec.Pod.TerminationGracePeriodSeconds
 	podSpec.ServiceAccountName = z.Spec.Pod.ServiceAccountName
+	if z.Spec.VolumePermissions == true && z.Spec.Pod.SecurityContext != nil {
+		runAsUser := int64(0)
```

Comment:
I think that both the RunAsUser field of the init container and the main container both need to be some specific value from the values file (such as "0" but not necessarily), or auto (meaning randomized by openshift).

Hardcoding the runAsUser field of the init container to 0 is probably not what we want ultimately.  Because that means, if the init container runs as root, that essentially elevates the need for the whole pod (both containers) to run as root, since for example in Openshift, you'd need to give permission to the service account that runs the pod (zookeeper) access to the anyuid SCC.

Looking at the PostgreSQL chart, it shows [here ](https://github.com/bitnami/charts/blob/master/bitnami/postgresql/templates/statefulset.yaml#L112-L114) how it assigns a fully specified security context with RunAsUser if specified, or with the RunAsUser omitted if set to auto.

That is for the initcontainer itself.    Additionally, the target chown user that is used by the script to change the volume permission, is also following the same logic.  

https://github.com/bitnami/charts/blob/master/bitnami/postgresql/templates/statefulset.yaml#L86-L88

It does make things a bit confusing to have 2 security contexts (one for the initcontainer, and one for the main container); although it also brings flexbility.

For example, you could be in **scenario 1**:

Say your target user for the main container is 1001.
The the PV that is mounted inside the init container requires the user to be root in order to be able the permissions (usually due to some NFS server side configuration).  

In that case, you'd set the initcontainer's security context with RunAsUser: 0, and the maincontainer with security container: 1001.

The initcontainer script runs as root, ans chowns the volume to 1001.

**Scenario 2:**

You want your target to be running as 1001, and PV's permissions don't require the init container to run as root.  You can then set both to a security context with the 1001 context.    The initcontainer script runs as 1001 and chowns the volume to 1001.

**Scenario 3:**

You don't care what the runasuser is, you're running on Openshift and the volume doesn't require you to be root in order to change the permissions.

In that case; you would set the RunAsUser to auto; for both init and main container.  And let Openshift compute a uid when deploying the pod.   

The init container runs the script as user 123456, and chowns the target mount point to 123456

Reply:
@sarlaccpit  Addressed the comments. Please have a look